tools/rosbag_storage/src/bag.cpp

```diff
@@ -444,11 +654,29 @@ void Bag::stopWritingChunk() {
 
     // Clear the connection counts
     curr_chunk_info_.connection_counts.clear();
-    
+
     // Flag that we're starting a new chunk
     chunk_open_ = false;
 }
 
+uint32_t Bag::encryptChunk(const uint32_t compressed_size) {
+    // Read existing (compressed) chunk
+    std::basic_string<unsigned char> compressed_chunk(compressed_size, 0);
+    seek(curr_chunk_data_pos_);
+    file_.read((char*) &compressed_chunk[0], compressed_size);
```

Comment:
Does this mean the data is first written compressed into the bag file, then read from the bag file, padded and encrypted then written back into the file? 

Can the chunk be written compressed and encrypted the first time?

Reply:
That is obviously more efficient, and requires refactoring around compressing classes.  I wanted to make this PR simple, and leave compression and encryption using memory as a future improvement.